### PR TITLE
Azure Bug Fix

### PR DESCRIPTION
**Pull request by mayur040993**
…are more than 1000

**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**
Node Attestation was failing due to pagination if VMs are more than 1000 in a particular subscription

**Which issue this PR fixes**

### CODE REVIEW

File: pkg/server/plugin/noderesolver/azure/client.go [hunk 1]
```diff
@@ -58,6 +58,13 @@ func (c *azureClient) GetVirtualMachineResourceID(ctx context.Context, principal
 	}
 
 	values := result.Values()
+	for len(values) == 0 {
```
Comment (rturner3) [inline pkg/server/plugin/noderesolver/azure/client.go:61]: This usage of the result doesn't look quite right. I would not expect the first page of values to be empty if there were multiple pages of results.

This also obscures the error case when no values are returned (handled in the next `if` block) by returning a more generic error from the Azure SDK from `result.NextWithContext()`.

**Review by rturner3 — COMMENTED** [review]
Hi @mayur040993 , would you mind elaborating a bit on how you are setting up system identity for the VMs running SPIRE Agent in your environment and the particular issue you are seeing?

It was unclear to me in what circumstances you might see multiple VMs with the same service principal ID in an Azure subscription. Just to clarify, SPIRE is designed to use system-assigned managed service identity tokens for `azure_msi` node attestation, as these system identities should be unique per VM instance. System-assigned service identities are used instead of user-assigned service identities because user-assigned identities are not guaranteed to be unique to a particular VM instance.

**Comment by azdagron** [comment]
@mayur040993, friendly ping on answering @rturner3's message above :)

**Comment by rturner3** [comment]
Hey @mayur040993 , we're going to close this PR for now since it has gone stale. If you discover any other issues around this code path, feel free to open another PR.

File: pkg/server/plugin/noderesolver/azure/client.go (hunk 1 above)
Comment (mayur040993) [reply pkg/server/plugin/noderesolver/azure/client.go:61]: Azure SDK sometimes gives you an empty response with page id if the num of the VMs is more than 1000.
so to handle this you have to traverse to different pages as well.

Currently, this is not taken care of by the Azure SDK.

Comment (mayur040993) [reply pkg/server/plugin/noderesolver/azure/client.go:61]: So if you are getting the next page id in the response, it means that there is some data on the next page.

Comment (azdagron) [reply pkg/server/plugin/noderesolver/azure/client.go:61]: @mayur040993 do you have a link to the relevant documentation from the Azure APIs that describe this behavior?

**Pull request closed without merging**